doc/src/advanced/custom_potential/external.md

```diff
@@ -0,0 +1,205 @@
+    /// Exponent of repulsive contribution
+    n: f64,
+    /// Exponent of attractive contribution
+    m: f64,
+    /// Energetic prefactor computed from the exponents and epsilon
+    prefac: f64,
+}
+```
+
+In the first two lines we define our imports from `Lumol`, following with our structure.
+
+Next, we implement a constructor function. That's usefull in this case since we want to compute the prefactor
+of the potential once before we start our simulation.
+
+$\text{prefac} = \frac{n}{n-m} \left(\frac{n}{m}\right)^{m/(n-m)}\epsilon$
+
+In Rust we typically use `new` for the constructors' name.
+
+```rust
+impl Mie {
+    pub fn new(sigma: f64, epsilon: f64, n: f64, m: f64) -> Mie {
+        if m >= n {
+            panic!("The repulsive exponent n has to be larger than the attractive exponent m")
+        };
+        let prefac = n / (n - m) * (n / m).powf(m / (n - m)) * epsilon;
+        Mie {
+            sigma: sigma,
+            epsilon: epsilon,
+            n: n,
+            m: m,
+            prefac: prefac,
+        }
+    }
+}
+```
+
+Our function takes the parameter set as input, computes the prefactor and returns a `Mie` struct. Notice that
+it panics, for `n` smaller than or equal to `m`.
+The next step is to implement the `Potential` trait for `Mie`.
+
+## Implementing `Potential`
+
+Add the following lines below the structs implementation.
+
+```rust
+impl Potential for Mie {
+    fn energy(&self, r: f64) -> f64 {
+        let sigma_r = self.sigma / r;
+        let repulsive = f64::powf(sigma_r, self.n);
+        let attractive = f64::powf(sigma_r, self.m);
+        self.prefac * (repulsive - attractive)
+    }
+
+    fn force(&self, r: f64) -> f64 {
+        let sigma_r = self.sigma / r;
+        let repulsive = f64::powf(sigma_r, self.n);
+        let attractive = f64::powf(sigma_r, self.m);
+        -self.prefac * (self.n * repulsive - self.m * attractive) / r
+    }
+}
+```
+
+`energy` is the implementation of the Mie potential equation shown above.
+`force` is the negative derivative of the energy with respect to the distance, `r`.
+To be more precise, the vectorial force can readily be computed by multiplying the
+result of `force` with the connection vector $\vec{r}$.
+
+The next step is to make our `Potential` usable in Lumol's algortihms to compute
+non-bonded energies and forces.
+Therefore, we have to implement the `PairPotential` trait.
+
+## Implementing `PairPotential`
+
+Let's inspect the [documentation of `PairPotential`](http://lumol.org/lumol/latest/lumol/energy/trait.PairPotential.html).
+
+```rust
+pub trait PairPotential: Potential + BoxClonePair {
+    fn tail_energy(&self, cutoff: f64) -> f64;
+    fn tail_virial(&self, cutoff: f64) -> f64;
+
+    fn virial(&self, r: &Vector3D) -> Matrix3 { ... }
+}
+```
+
+First, we can see that `PairPotential` enforces the implementation of `Potential` which is denoted by `pub trait PairPotential: Potential ...` (we ignore `BoxClonePair` for now).
```

Comment:
Maybe we can add a comment saying that `BoxClonePair` is automatically implemented for us if we implement `PairPotential` manually?